CellFabric/cell_fabric/pdk.py

```diff
@@ -143,4 +146,28 @@ def get_via_stack(self):
         return layer_stack
 
     def get_gds_map(self):
-        return {x: self.pdk[x]['LayerNo'] for x in self.pdk.keys() if 'LayerNo' in self.pdk[x]}
\ No newline at end of file
+        return {x+y: [self.pdk[x]['GdsLayerNo'], self.pdk[x]['GdsDatatype'][y]] for x in self.pdk.keys() if 'GdsLayerNo' in self.pdk[x] for y in self.pdk[x]['GdsDatatype']}
```

Comment:
Looks good overall. Couple of things to address before merging.
1) Fix CI issues (I think you are already working on this)
2) Don't create multiple keys for the same layer as you have done in line 149 of pdk.py. You are doing the x+y key creation both over here & in gen_gds_json. Its better to keep the original dict intact & explicitly reference LayerNo & DataType when needed.

Reply:
Thanks for the comments!
I will fix these issues. 
Could you please check "primitive.py" (placement of body contact part)? Whether it is correct or not.

Reply:
> Thanks for the comments!
> I will fix these issues.
> Could you please check "primitive.py" (placement of body contact part)? Whether it is correct or not.

Did a more detailed review. Let me know if something seems unclear. I can get in there and refactor stuff myself. Just didn't want us to be stepping on each other's commits.